### PR TITLE
Add support for Scala 3.0.0-RC1

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

**Comment by tgodzik** [comment]
Surprisingly enough, everything seems to work this time :sweat_smile: 

I will take a look at the pending PRs and try to merge them before the release. We should also update scalameta after a new release, we should be able to parse most existing Scala 3 code now.

**Comment by ckipp01** [comment]
Alright so I haven't actually dug into the Bloop commits, but testing out the last couple Bloop versions, the last one that worked for me was `1.4.6-35-09d45cbd`. Starting with `1.4.6-38-2f6fdcba` (and when I test this pr) I get the following when it attempts to make the BSP connection with Bloop.


```
Waiting for the bsp connection to come up...
error: The command bsp --protocol local --socket /var/folders/fq/nx_jsnyd6550xp03czx898d40000gn/T/bsp-launcher6777020916520002691/bsp.socket returned with an error
> No server running at 127.0.0.1:8212, let's fire one...
> Resolving ch.epfl.scala:bloop-frontend_2.12:1.4.6-38-2f6fdcba...
> Starting bloop server at 127.0.0.1:8212...
> Attempting a connection to the server...
> Attempting a connection to the server...
> Attempting a connection to the server...
> Attempting a connection to the server...
> Attempting a connection to the server...
> error: Unexpected error forces client exit!
> java.io.EOFException
>       at java.base/java.io.DataInputStream.readInt(DataInputStream.java:397)
>       at bloop.shaded.snailgun.protocol.Protocol.$anonfun$processChunkFromServer$1(Protocol.scala:181)
>       at scala.util.Try$.apply(Try.scala:213)
>       at bloop.shaded.snailgun.protocol.Protocol.processChunkFromServer(Protocol.scala:180)
>       at bloop.shaded.snailgun.protocol.Protocol.sendCommand(Protocol.scala:108)
>       at bloop.shaded.snailgun.TcpClient.run(TcpClient.scala:34)
>       at bloop.bloopgun.BloopgunCli.executeCmd$1(Bloopgun.scala:268)
>       at bloop.bloopgun.BloopgunCli.fireCommand(Bloopgun.scala:286)
>       at bloop.bloopgun.BloopgunCli.run(Bloopgun.scala:230)
>       at bloop.launcher.bsp.BspBridge$$anon$1.run(BspBridge.scala:82)
error: Trying a tcp-based connection to the server instead...
Opening a bsp server connection with 'bsp --protocol tcp --port 34153'...
Waiting for the bsp connection to come up...
error: The command bsp --protocol tcp --port 34153 returned with an error
> error: Unexpected error forces client exit!
> java.io.EOFException
>       at java.base/java.io.DataInputStream.readInt(DataInputStream.java:397)
>       at bloop.shaded.snailgun.protocol.Protocol.$anonfun$processChunkFromServer$1(Protocol.scala:181)
>       at scala.util.Try$.apply(Try.scala:213)
>       at bloop.shaded.snailgun.protocol.Protocol.processChunkFromServer(Protocol.scala:180)
>       at bloop.shaded.snailgun.protocol.Protocol.sendCommand(Protocol.scala:108)
>       at bloop.shaded.snailgun.TcpClient.run(TcpClient.scala:34)
>       at bloop.bloopgun.BloopgunCli.executeCmd$1(Bloopgun.scala:268)
>       at bloop.bloopgun.BloopgunCli.fireCommand(Bloopgun.scala:274)
>       at bloop.bloopgun.BloopgunCli.run(Bloopgun.scala:230)
>       at bloop.launcher.bsp.BspBridge$$anon$1.run(BspBridge.scala:82)
error: The launcher failed to establish a bsp connection, aborting...
2021.02.18 09:11:16 ERROR Failed to connect with build server, no functionality will work.
java.lang.RuntimeException: The server did not start, got FailedToOpenBspConnection
        at bloop.launcher.LauncherMain.failPromise$1(Launcher.scala:95)
        at bloop.launcher.LauncherMain.runLauncher(Launcher.scala:122)
        at scala.meta.internal.metals.BloopServers$$anon$1.run(BloopServers.scala:159)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:834)
```
Even trying a `bloop about` at this point then fails:

```
❯ bloop about
error: Unexpected error forces client exit!
java.io.EOFException
        at java.io.DataInputStream.readInt(DataInputStream.java:397)
        at snailgun.protocol.Protocol.$anonfun$processChunkFromServer$1(Protocol.scala:181)
        at scala.util.Try$.apply(Try.scala:213)
        at snailgun.protocol.Protocol.processChunkFromServer(Protocol.scala:180)
        at snailgun.protocol.Protocol.sendCommand(Protocol.scala:108)
        at snailgun.TcpClient.run(TcpClient.scala:34)
        at bloop.bloopgun.BloopgunCli.executeCmd$1(Bloopgun.scala:268)
        at bloop.bloopgun.BloopgunCli.fireCommand(Bloopgun.scala:274)
        at bloop.bloopgun.BloopgunCli.run(Bloopgun.scala:230)
        at bloop.bloopgun.Bloopgun$.main(Bloopgun.scala:638)
        at bloop.bloopgun.Bloopgun.main(Bloopgun.scala)
```

**Review by ckipp01 — APPROVED** [review]
For now this makes sense until we figure out what's going on with Bloop.

**Comment by tgodzik** [comment]
I actually bumped the version to 1.4.8 Bloop with the fix, so we should be good to merge it.

File: tests/cross/src/test/scala/tests/pc/CompletionSuite.scala [hunk 1]
```diff
@@ -426,7 +426,8 @@ class CompletionSuite extends BaseCompletionSuite {
       |""".stripMargin,
     "",
     compat = Map(
-      "3.0.0-M3" -> "Inner: a.Outer.Inner$"
+      "3.0.0-M3" -> "Inner: a.Outer.Inner$",
+      "3.0.0-RC1" -> "Inner: a.Outer.Inner$"
```
Comment (gabro) [inline tests/cross/src/test/scala/tests/pc/CompletionSuite.scala:430]:
```suggestion
      "3.0" -> "Inner: a.Outer.Inner$",
```

?

**Review by gabro — APPROVED** [review]
Code looks 👍  (just one non-blocking suggestion)

File: tests/cross/src/test/scala/tests/pc/CompletionSuite.scala (hunk 1 above)
Comment (tgodzik) [reply tests/cross/src/test/scala/tests/pc/CompletionSuite.scala]: done!

**Pull request merged**